routes.py

```diff
@@ -28,36 +28,41 @@ def list_cars():
     if cars:
         return jsonify({'cars': list_of_cars})
     else:
-        return {'message': "You don't have any cars in your parking lot. Just add some cars"}
+        return {'status': 'error', 'message': "You don't have any cars in your parking lot. Just add some cars"}
 
 
 @app.route('/add', methods=['GET', 'POST'])
 def add_car():  # in browser URL : http://127.0.0.1:7000/add?car_number=CJ45WAY&tariff=hourly
 
-    car_number = request.args.get('car_number')
+    new_car_number = request.args.get('car_number')
     tariff = request.args.get('tariff')
 
+    # Check if there is no car in the parking with the same car_number:
+    existing_car_number_count = Parking.query.filter_by(car_number=new_car_number).count()
+    if existing_car_number_count > 0:
+        return {'status': 'error', 'message': 'Hey! There is already a car with that number. I will call the Police !'}
+
     # Check that the tariff is only hourly or daily :
     if tariff not in ['hourly', 'daily']:
-        raise NameError('the tariff must be either hourly or daily')
+        return {'status': 'error', 'message': 'the tariff must be either hourly or daily'}
 
-    # Check whether a car_number is alphanumeric
-    if not str(car_number).isalnum() :
-        raise  NameError('the car_number must be compose only of letters and digits')
+    # Check whether a new_car_number is alphanumeric
+    if not str(new_car_number).isalnum():
+        raise {'status': 'error', 'message': 'the new_car_number must be compose only of letters and digits'}
 
     # Also to note here that for further checks we must use a URL Validator as if one types :
     # http://127.0.0.1:7000/add?car_number=CT33M&OM&tariff=hourly a car will be added because
-    #  &OM string will be ignored as it is undefined. and the car_number=CT33M will be added instead
+    #  &OM string will be ignored as it is undefined. and the new_car_number=CT33M will be added instead
 
     # Check if there are enough available places in the Parking :
     if Settings.available_places > 0:
         # Form the object of the car :
-        new_car = Parking(car_number=car_number, tariff=tariff)
+        new_car = Parking(car_number=new_car_number, tariff=tariff)
         try:
             db.session.add(new_car)
             db.session.commit()
-            Settings.available_places -= 1      # Update available places
-            return {'status': 'success', 'id': new_car.id, 'car_number': new_car.car_number,
+            Settings.available_places -= 1  # Update available places
+            return {'status': 'success', 'id': new_car.id, 'new_car_number': new_car.car_number,
                     'tariff': new_car.tariff, 'date_start': new_car.date_start,
                     'available_places': Settings.available_places}
         except Exception:
```

Comment:
'there was an error adding your car' how to investigate such failures? Let's make it clear to don't overload our support with customer's calls

Reply:
Also, shouldn't result be like "{'status': 'error',...}"?